src/components/views/settings/account/EmailAddresses.js

```diff
@@ -193,8 +193,14 @@ export default class EmailAddresses extends React.Component {
             this.props.onEmailsChange(emails);
         }).catch((err) => {
             this.setState({continueDisabled: false});
-            if (err.errcode !== 'M_THREEPID_AUTH_FAILED') {
-                const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
+            const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
+            if (err.errcode === 'M_THREEPID_AUTH_FAILED') {
```

Comment:
Maybe `matrix-react-sdk/src/components/views/settings/discovery/EmailAddresses.js` (the discovery variant) should get the same treatment, but it's fine to save to a separate PR as we should get the bug fix landed.

Reply:
Hmm, that code looks almost entirely duplicated :(

Have put the dialog there as well in any case.

Reply:
Yes, it is mostly duplicated. 😭